src/lib.rs

```diff
@@ -1,4 +1,77 @@
-//! The [Neon](https://www.neon-bindings.com/) crate provides bindings for writing Node.js plugins with a safe and fast Rust API.
+//! The [Neon](https://www.neon-bindings.com/) crate provides bindings for
+//! writing [Node.js addons](https://nodejs.org/api/addons.html) (i.e.,
+//! dynamically-linked binary modules) with a safe and fast Rust API.
+//!
+//! ## Getting Started
+//!
+//! You can conveniently bootstrap a new Neon project with the Neon project
+//! generator. You don't need to install anything special on your machine as
+//! long as you have a supported version of Node on your system.
+//!
+//! To start a new project, open a terminal in the directory where you would
+//! like to place the project, and run at the command prompt:
+//!
+//! ```text
+//! % npm init neon my-project
+//! ... answer the user prompts ...
+//! ✨ Created Neon project `my-project`. Happy 🦀 hacking! ✨
+//! ```
+//!
+//! where `my-project` can be any name you like for the project. This will
+//! run the Neon project generator, prompting you with a few questions and
+//! placing a simple but working Neon project in a subdirectory called
+//! `my-project` (or whatever name you chose).
+//!
+//! You can then install and build the project by changing into the project
+//! directory and running the standard Node installation command:
+//!
+//! ```text
+//! % cd my-project
+//! % npm install
+//! % node
+//! > require(".").hello()
+//! 'hello node'
+//! ```
+//!
+//! You can look in the project's generated `README.md` for more details on
+//! the project structure.
+//!
+//! ## Example
+//!
+//! The generated `src/lib.rs` contains a function annotated with the
+//! `#[neon::main]` attribute, marking it as the module's main entry point
+//! to be executed when the module is first loaded. This function can have
+//! any name but is conventionally called `main`:
+//!
+//! ```
+//! # use neon::prelude::*;
+//! #
+//! # fn hello(mut cx: FunctionContext) -> JsResult<JsString> {
+//! #    Ok(cx.string("hello node"))
+//! # }
+//! #
+//! #[neon::main]
+//! fn main(cx: ModuleContext) -> NeonResult<()> {
+//!     cx.export_function("hello", hello)?;
+//!     Ok(())
+//! }
+//! ```
+//!
+//! The example code generated by `neon init neon` exports a single
+//! function via [`ModuleContext::export_function`](./context/struct.ModuleContext.html#method.export_function).
+//! The `hello` function is defined just above `main` in `src/lib.rs`:
+//!
+//! ```
+//! # use neon::prelude::*;
+//! #
+//! fn hello(mut cx: FunctionContext) -> JsResult<JsString> {
+//!     Ok(cx.string("hello node"))
+//! }
+//! ```
+//!
+//! The `hello` function takes a [`FunctionContext`](./context/type.FunctionContext.html) and
```

Comment:
Rust doc was recently updated to support linking by name.

```suggestion
//! The `hello` function takes a [`FunctionContext`](context::FunctionContext) and
```

This is more robust against HTML structure changes. There are also shorthand versions when the name and display text are the same.

https://doc.rust-lang.org/rustdoc/linking-to-items-by-name.html

*Note*: This feedback applies to multiple places in these docs, but those are omitted for brevity.

Reply:
Oh I was hoping that would be the case, awesome!